src/components/form/index.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useForm, Controller } from 'react-hook-form';
+
+const Form = props => {
+  const {
+    defaultValues,
+    children,
+    onSubmit,
+    validationResolver,
+    ...rest
+  } = props;
+
+  const methods = useForm({ defaultValues, validationResolver });
+  const { handleSubmit } = methods;
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} {...rest}>
+      {Array.isArray(children)
+        ? children.map(child => {
```

Comment:
I'm under the assumption that this is suppose to work as:
```javascript
<Form>
   <Field>
</Form>
```

If so - I think it would be more readable if we just wrote it as such instead of this abstraction.

Reply:
I am not sure what do you mean by how to make it more readable. Do you mean instead passing Form inside Box component simply use <Form><Field/><Form> ?

It's actually smart form component: https://react-hook-form.com/advanced-usage/#SmartFormComponent

Reply:
Thanks for clarifying --  I didn't catch that this came from the SmartFoormComponent the first time 😄